website/project/model.py

```diff
@@ -84,36 +83,32 @@ class MetaSchema(StoredObject):
     schema = fields.DictionaryField()
     category = fields.StringField()
 
-    # Version of the Knockout metadata renderer to use (e.g. if data binds
-    # change)
+    # Deprecated legacy field
     metadata_version = fields.IntegerField()
     # Version of the schema to use (e.g. if questions, responses change)
     schema_version = fields.IntegerField()
 
+def ensure_schema(schema, name, version=1):
+    try:
+        schema_obj = MetaSchema.find_one(
+            Q('name', 'eq', name) &
+            Q('schema_version', 'eq', version)
+        )
+    except NoResultsFound:
+        meta_schema = {
+            'name': name,
+            'schema_version': version,
+            'schema': schema,
+        }
+        schema_obj = MetaSchema(**meta_schema)
+        schema_obj.save()
+    return schema_obj
 
-def ensure_schemas(clear=True):
-    """Import meta-data schemas from JSON to database, optionally clearing
-    database first.
-
-    :param clear: Clear schema database before import
+def ensure_schemas():
+    """Import meta-data schemas from JSON to database if not already loaded
     """
-    if clear:
-        try:
-            MetaSchema.remove()
-        except AttributeError:
-            if not settings.DEBUG_MODE:
-                raise
     for schema in OSF_META_SCHEMAS:
-        try:
-            MetaSchema.find_one(
-                Q('name', 'eq', schema['name']) &
-                Q('schema_version', 'eq', schema['schema_version'])
-            )
-        except:
-            schema['name'] = schema['name'].replace(' ', '_')
-            schema_obj = MetaSchema(**schema)
-            schema_obj.save()
-
+        ensure_schema(schema, schema['name'], schema.get('version', 1))
```

Comment:
As APIs evolve, it's not uncommon to add new kwargs or reorganize the order of arguments; function authors tend to assume that the order (or presence) of a given kwarg shouldn't matter, because it is always bound by name anyway.

If you treat a kwarg as positional, suddenly that assumption breaks down, and your code stops working for no good reason because of external cleanup. It's the kind of bug that bites a lot of libraries, but can be prevented with minimal effort.

It's not just me: The authors of python felt strongly enough about enforcing keyword usage that they actually introduced *keyword-only* args in python3.